h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+    if isinstance(map_preferences, dict):
+        map_preferences = GeospatialMapConfig.from_dict(map_preferences, strict=True)
+
+    # Load data
+    shipping_coords_fpath = find_file(shipping_coords_fpath)
+
+    # If case_results_fpath is a .csv file read in with pandas
+    if (".csv") in shipping_coords_fpath.suffix:
+        shipping_coords_df = pd.read_csv(shipping_coords_fpath, index_col=0)
+
+    else:
+        raise TypeError(
+            f"The provided filepath {shipping_coords_fpath} is of the wrong type, must be a .csv"
+        )
+
+    # Auto detect latitude and longitude column names if not provided as argument
+    if latitude_var_name is None:
+        latitude_var_name, _ = auto_detect_lat_long_columns(shipping_coords_df)
+
+    if longitude_var_name is None:
+        _, longitude_var_name = auto_detect_lat_long_columns(shipping_coords_df)
+
+    # Order columns so tuples created in dict below are (long,lat) ordered pairs
+    shipping_coords_df = shipping_coords_df[[longitude_var_name, latitude_var_name]]
+    shipping_coords_dict = {
+        key: tuple(value.values())
+        for key, value in shipping_coords_df.to_dict(orient="index").items()
+    }
+
+    # Create list of coordinates to trace shipping route
+    shipping_route_coords = [shipping_coords_dict[str(city)] for city in shipping_route]
+
+    # Create GeoDataFrame with shipping_route_coords
+    shipping_route_gdf = gpd.GeoDataFrame(
+        geometry=[LineString(shipping_route_coords)], crs=map_preferences.lat_long_crs
+    )
+
+    # Convert coordinates to typical Web Mercator project CRS (EPSG:3857) for plotting
+    shipping_route_gdf = shipping_route_gdf.to_crs(map_preferences.web_map_crs)
+
+    # Create list variable to hold all gdfs for calculating basemap bounds
+    gdfs_for_bounds = [shipping_route_gdf]
+
+    # Check if fig, ax, and base_layer_gdf are all None
+    if all(v is None for v in [fig, ax, base_layer_gdf]):
+        fig, ax = plt.subplots(
+            1,
+            figsize=map_preferences.figsize,
+            constrained_layout=map_preferences.constrained_layout,
+        )
+
+    # Else if fig, ax, and base_layer_gdf are all not None
+    elif all(v is not None for v in [fig, ax, base_layer_gdf]):
+        # set figure and axes as current fig,ax objects
+        plt.figure(fig.number)
+        plt.sca(ax)
+        # Validate base_layer_gdf(s) is in the same CRS as the results_gdf
+        base_layer_gdf = validate_gdfs_are_same_crs(base_layer_gdf, shipping_route_gdf)
+        # Extend gdf_for_bounds with base_layer_gdf(s) provided
+        gdfs_for_bounds.extend(base_layer_gdf)
+
+    # Else fig, ax, and base_layer_gdf not provided as a full set
+    else:
+        raise ValueError(
+            """The fig, ax, and base_layer_gdf arguments must be provided together to add a layer to
+            an existing plot or all must be omitted/None to create a new plot"""
+        )
```

Comment:
This is another spot where the check for the presence of all values or none of the values being `None` could be simplified 

```suggestion
    # Ensure all of fig, ax, and base_layer_gdf are None or have a value provided, not mixed
    _check = [el is None for el in [fig, ax, base_layer_gdf]]
    if all(_check):
        fig, ax = plt.subplots(
            1,
            figsize=map_preferences.figsize,
            constrained_layout=map_preferences.constrained_layout,
        )
    elif any(_check):
        msg = (
            "The fig, ax, and base_layer_gdf arguments must be provided together to add a layer to"
            "an existing plot or all must be omitted/None to create a new plot"
        )
        raise ValueError(msg)
    else:
        plt.figure(fig.number)
        plt.sca(ax)
        base_layer_gdf = validate_gdfs_are_same_crs(base_layer_gdf, shipping_route_gdf)
        gdfs_for_bounds.extend(base_layer_gdf)
```